dataflowlauncher/launcher.py

```diff
@@ -40,10 +40,16 @@ def run(args, unknown_args, exec_path):
 
     """Generate parameters with config readers, update parameters with cli parsers """
     parameter_dict = get_jar_parameter_dict(config)
-    parameter_dict.update(get_updated_launch_params(config, args))
 
-    parameter_list = get_formatted_launch_parameters(parameter_dict, args.unknown_arguments, unknown_args)
-    print_launch_parameters(parameter_list)
+    updated_parameters = get_updated_launch_params(parameter_dict, args)
+
+    updated_parameter_dict = dict(parameter_dict)
```

Comment:
Isn't `parameter_dict` already a dict? Does it need to be dicted again?

Reply:
This makes a copy of `parameter_dict` so that we can print out the difference between the old and new parameters

Reply:
Oh, missed that. Cool.